### ISSUE TITLE
Better maintenance of copyright date in sphinx refman

### ISSUE THREAD

**Issue by Alizter** [coq/coq#14628]
As discussed in #14620, perhaps it would be a good idea to have a current year variable in `doc/sphinx/conf.py` that can make changing the year more rigid. We can also add a test against the current year which can emit a warning if it differs.

That way, updated the copyright year can be a bit easier to remember.

cc @Zimmi48 @cpitclaudel

**Comment by cpitclaudel** [comment]
Sure, that seems reasonable, feel free to make a PR.

**Comment by Alizter** [comment]
It seems the main blocker to creating an automatic date is #7378. The issues with nix are apparently known in the sphinx community. https://github.com/sphinx-doc/sphinx/issues/3451

There is is recommended to change the `SOURCE_DATE_EPOCH` to be the date of the last commit: `$(shell git log -1 --format=%ct)`. We could probably hook this up in dune, though I am not sure how to pass around the output of git in dune.

There is also the question of "is this really an issue we need to fix". Since the refman on the website is not built with nix, is it an issue if a few users have 1970 as the copyright date? Does it really mean that much? If the user is building the documentation themselves then they could change the date to whatever they wish anyway.

In summary, I think a simple datetime call in the python config script should be enough and that #7378 is not really an issue worth fixing.

**Comment by Zimmi48** [comment]
> There is also the question of "is this really an issue we need to fix". Since the refman on the website is not built with nix, is it an issue if a few users have 1970 as the copyright date? Does it really mean that much? If the user is building the documentation themselves then they could change the date to whatever they wish anyway.

No, I don't think that's a big deal and we can leave this issue unfixed. However, I feel like nobody has really got to the bottom of the issue here, but I will reply about this in #7378 to keep the discussion on-topic.

**Comment by cpitclaudel** [comment]
I posted a comment there.  We can disable that behavior with `sphinx.config.correct_copyright_year = lambda *args, **kwargs: None`, I think (untested).

**Comment by ejgallego** [comment]
> There is is recommended to change the `SOURCE_DATE_EPOCH` to be the date of the last commit: `$(shell git log -1 --format=%ct)`. We could probably hook this up in dune, though I am not sure how to pass around the output of git in dune.

As discussed privately, something such as
```
(rule
  (targets source_date_epoch)
  (action (with-stdout-to %{targets} (run git ...))))
```
should do the trick, tho you likely want to have a script to handle the case git is not around, etc...

**Comment by Zimmi48** [comment]
That's not the right fix. The one proposed by Clément above is better since it would disable the misfeature from Sphinx. Anyway, that's also not the right issue to discuss this.
